packages/cmr-client/src/getUrl.ts

```diff
@@ -1,56 +1,67 @@
-function getCmrHost(cmrEnvironment?: string): string {
+/**
+ * Get host to use for CMR requests.
+ *
+ * @param {string} [cmrEnvironment]
+ *  CMR environment to use for requests. Can be a CMR environment "logical"
+ *  name or a full custom host name.
+ * @returns {string} - CMR host
+ * @private
+ **/
+export function getCmrHost(
+  cmrEnvironment: string | undefined = process.env.CMR_ENVIRONMENT
+): string {
+  if (!cmrEnvironment) {
+    throw new TypeError('CMR environment must be defined');
+  }
+  let cmrHost;
   switch (cmrEnvironment) {
     case 'OPS':
-      return 'cmr.earthdata.nasa.gov';
+      cmrHost = 'cmr.earthdata.nasa.gov';
+      break;
     case 'UAT':
-      return 'cmr.uat.earthdata.nasa.gov';
+      cmrHost = 'cmr.uat.earthdata.nasa.gov';
+      break;
     case 'SIT':
-      return 'cmr.sit.earthdata.nasa.gov';
+      cmrHost = 'cmr.sit.earthdata.nasa.gov';
+      break;
     default:
-      throw new TypeError(`Invalid CMR environment: ${cmrEnvironment}`);
+      cmrHost = cmrEnvironment;
```

Comment:
I'm slightly concerned that  the dual use of this both in it's variable form in the terraform and in this file are confusing - for example using `sit` versus `SIT` now would result in an attempt to connect to the url https://sit instead of an error.   

`Custom host or logical environment` != `cmrEnvironment`

Is there a reason to overload this at all, given the `host`setup being removed from the relevant functions?    Meaning set a seperate environment variable (e.g. cmr_custom_host), and either explicitly return if it's set and cmr_environment = 'CUSTOM' or something like that? 

Else could we find a better name for the overloaded variable (and I dislike writing this without a suggestion)?

Reply:
I agree with the concern.

The reason for the overload was that if you have two variables, then you have to decide the preference between the two and note that. I assume that the `cmr_custom_host` would be preferred over the `cmr_environment`. I had started to implement it that way but for some reason wasn't liking it. But I think you're right that it is better to have strict validation on the `cmr_environment`

Reply:
Updates in 4ac10776f3050fe370f80202a8bc4605441666dd look good.   👍🏻